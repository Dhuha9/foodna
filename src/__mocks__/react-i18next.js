const React = require('react');

const useMock = [(k) => k, {}];
useMock.t = (k) => k;
useMock.i18n = {};

module.exports = {
  // this mock makes sure any components using the translate HoC receive the t function as a prop
  // eslint-disable-next-line react/display-name
  withTranslation: () => (Component) => (props) => (
    <Component t={(k) => k} {...props} />
  ),

  useTranslation: () => useMock,
};
