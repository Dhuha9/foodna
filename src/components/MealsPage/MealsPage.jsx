/* eslint react/prop-types: 0 */
/* eslint-disable-next-line react/jsx-no-bind */
import React, { Component } from 'react';
import MealCard from '../MealCard/MealCard';
//import MealData from '../../data.json'
import firebase from '../../firebase'


export default class MealsPage extends Component {
  
  constructor(){
    super()
    this.state = {
      meals:[]
    }
  }

  componentDidMount(){
      const unsubscribe = firebase.firestore()
      .collection('meals')
      .onSnapshot((snap)=>{
        const newMeals = snap.docs.map((doc)=>({
          id: doc.id,
          ...doc.data()
        }))
        this.setState({
          meals: newMeals
        })
      })
      return()=> unsubscribe();
  }
  


 

  render() {
    return (
      <div className="available-meals-page">
        <h1 className="available-meals-title">Available Meals Right Now</h1>
        <p className="available-meals-subtitle">
          If you dont try this app you will never become the superhero you were
          meant to be
        </p>
        <div className="meals-grid">
            {this.state.meals.map((meal)=>{
                return <div key={meal.id}>
                  <MealCard 
                img={meal.image}
                title={meal.title}
                organization={meal.organization}
                />
                  </div>
            })}
        </div>
      </div>
    );
  }
}
