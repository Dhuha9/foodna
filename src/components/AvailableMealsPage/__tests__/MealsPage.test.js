import * as React from 'react';
import * as renderer from 'react-test-renderer';
import MealsPage from '../MealsPage';
import { withTranslation } from 'react-i18next';

it('renders apply now correctly', () => {
  const TranslatedMealsPage = withTranslation()(MealsPage);

  const tree = renderer.create(<TranslatedMealsPage />).toJSON();
  expect(tree).toMatchSnapshot();
});
