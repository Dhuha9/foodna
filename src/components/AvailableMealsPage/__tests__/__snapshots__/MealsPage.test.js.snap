// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders apply now correctly 1`] = `
<div
  className="available-meals-page"
>
  <h1
    className="available-meals-title text-center"
  >
    AvailableMealsComponent.headline
  </h1>
  <p
    className="available-meals-subtitle text-center"
  >
    AvailableMealsComponent.paragraph
  </p>
  <div
    className="meals-grid"
  />
</div>
`;
